react/features/recording/components/Recording/StartRecordingDialog.js

```diff
@@ -0,0 +1,231 @@
+    }
+
+    /**
+     * Validates the oauth access token.
+     *
+     * @inheritdoc
+     * @returns {void}
+     */
+    componentDidMount() {
+        if (typeof this.props._token !== 'undefined') {
+            this._onTokenUpdated();
+        }
+    }
+
+    /**
+     * Validates the oauth access token.
+     *
+     * @inheritdoc
+     * @returns {void}
+     */
+    componentDidUpdate(prevProps) {
+        if (this.props._token !== prevProps._token) {
+            this._onTokenUpdated();
+        }
+    }
+
+    /**
+     * Validates the dropbox access token and fetches account information.
+     *
+     * @returns {void}
+     */
+    _onTokenUpdated() {
+        const { _clientId, _token } = this.props;
+
+        if (typeof _token === 'undefined') {
+            this.setState({
+                isTokenValid: false,
+                isValidating: false
+            });
+        } else {
+            this.setState({
+                isTokenValid: false,
+                isValidating: true
+            });
+            getDropboxData(_token, _clientId).then(data => {
+                if (typeof data === 'undefined') {
+                    this.setState({
+                        isTokenValid: false,
+                        isValidating: false
+                    });
+                } else {
+                    this.setState({
+                        isTokenValid: true,
+                        isValidating: false,
+                        ...data
+                    });
+                }
+            });
+        }
+    }
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+        const { isTokenValid, isValidating, spaceLeft, userName } = this.state;
+
+        return (
+            <Dialog
+                okDisabled = { !isTokenValid }
+                okTitleKey = 'dialog.confirm'
+                onSubmit = { this._onSubmit }
+                titleKey = 'dialog.recording'
+                width = 'small'>
+                <StartRecordingDialogContent
+                    isTokenValid = { isTokenValid }
+                    isValidating = { isValidating }
+                    spaceLeft = { spaceLeft }
+                    userName = { userName } />
+            </Dialog>
+        );
+    }
+
+    _onSubmit: () => boolean;
+
+    /**
+     * Starts a file recording session.
+     *
+     * @private
+     * @returns {boolean} - True (to note that the modal should be closed).
+     */
+    _onSubmit() {
+        sendAnalytics(
+            createRecordingDialogEvent('start', 'confirm.button')
+        );
+        const { _conference, _token } = this.props;
+
+        _conference.startRecording({
+            mode: JitsiRecordingConstants.mode.FILE,
+            appData: JSON.stringify({
+                'file_recording_metadata': {
+                    'upload_credentials': {
+                        'service_name': 'dropbox',
```

Comment:
I guess when we add second service we will fix this problem. Don't you think it's better to leave it like it is for now instead of planning what would be the requirements for the second service and how to meet them?